### PR TITLE
Typo fixes in ReaderWriterLockSlim.cs comments

### PR DESCRIPTION
**Pull request by rcabr**
Minor fixes to correct typos in comments.

### CODE REVIEW

File: src/System.Threading/src/System/Threading/ReaderWriterLockSlim.cs [hunk 1]
```diff
@@ -55,12 +55,12 @@ public class ReaderWriterLockSlim : IDisposable
         //Specifying if locked can be reacquired recursively.
         private bool _fIsReentrant;
 
-        // Lock specifiation for myLock:  This lock protects exactly the local fields associted
-        // instance of ReaderWriterLockSlim.  It does NOT protect the memory associted with the
+        // Lock specifiation for myLock:  This lock protects exactly the local fields associated
```
Comment (stephentoub) [inline src/System.Threading/src/System/Threading/ReaderWriterLockSlim.cs]: Should this be "associated with this" rather than just "associated"?

**Comment by stephentoub** [comment]
One comment, otherwise LGTM.  Thanks for the cleanup.

File: src/System.Threading/src/System/Threading/ReaderWriterLockSlim.cs (hunk 1 above)
Comment (sharwell) [inline src/System.Threading/src/System/Threading/ReaderWriterLockSlim.cs]: :bulb: specifi*c*ation

:thought_balloon: Honestly, it seems everything before the `:` could be omitted.

File: src/System.Threading/src/System/Threading/ReaderWriterLockSlim.cs [hunk 2]
```diff
@@ -55,12 +55,12 @@ public class ReaderWriterLockSlim : IDisposable
...
+        // instance of ReaderWriterLockSlim.  It does NOT protect the memory associated with the
```
Comment (sharwell) [inline src/System.Threading/src/System/Threading/ReaderWriterLockSlim.cs]: :bulb: ~~the~~ the

(There's a duplicated word at the end of the line)

**Comment by sharwell** [comment]
Two comments, otherwise :+1:

**Comment by rcabr** [comment]
Thanks for the feedback. Updating with your corrections and suggestions.

**Comment by stephentoub** [comment]
@rcabr, thanks, but you don't need to close the PR.  You can simply update your commit, and the PR will automatically reflect that.

**Comment by rcabr** [comment]
My bad.  Will reopen and update. Thanks!

**Comment by stephentoub** [comment]
Thanks.  LGTM.  Could you squash your commits so that there's just one instead of two?  Then we can merge once the CI build completes.

**Comment by rcabr** [comment]
Squashed commits into a single commit.

**Comment by stephentoub** [comment]
Thanks, @rcabr.  I'll merge this once it passes the CI build.

**Pull request merged**